### ISSUE TITLE
Is useImperativeHandle redundant?

### ISSUE THREAD

**Issue by evoyy** [facebook/react#16530]
Excuse my naivety, but I see no need for useImperativeHandle. Instead of this: 

````javascript
// https://reactjs.org/docs/hooks-reference.html#useimperativehandle

function FancyInput(props, ref) {
  const inputRef = useRef();
  useImperativeHandle(ref, () => ({
    focus: () => {
      inputRef.current.focus();
    }
  }));
  return <input ref={inputRef} ... />;
}
FancyInput = forwardRef(FancyInput);
````

Why not simply assign the current property to the forwarded ref?

````javascript
function FancyInput(props, fwdRef) {
  const inputRef = useRef();
  useEffect(() => {
    if (fwdRef) {
      fwdRef.current = inputRef.current;
    }
  }, []);
  return <input ref={inputRef} ... />;
}
FancyInput = forwardRef(FancyInput);
````

**Comment by jquense** [comment]
It is mostly redundant in this example yes. But you expose more than the same methods vai the hook.

consider something like:

```js
 useImperativeHandle(ref, () => ({
     scrollToBottom: () => {
          listRef.current!.scrollTop = listRef.current!.scrollHeight;
     },
  }));

return <ScrollableList ref={ref} />
```

**Comment by evoyy** [comment]
Interesting, thanks. Closing issue.

**Comment by Job456** [comment]
Lol

On Wed, Aug 21, 2019 at 2:12 PM evoyy <notifications@github.com> wrote:

> Closed #16530.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
-- 
Ipad

**Comment by liyan1105** [comment]
> It is mostly redundant in this example yes. But you can expose more than the same methods via the hook.
> 
> consider something like:
> 
> ```js
>  useImperativeHandle(ref, () => ({
>      scrollToBottom: () => {
>           listRef.current!.scrollTop = listRef.current!.scrollHeight;
>      },
>   }));
> 
> return <ScrollableList ref={ref} />
> ```

useEffect also can do that.
